controller/konnect/ops_controlplane.go

```diff
@@ -0,0 +1,165 @@
+package konnect
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	sdkkonnectgo "github.com/Kong/sdk-konnect-go"
+	"github.com/Kong/sdk-konnect-go/models/components"
+	"github.com/Kong/sdk-konnect-go/models/sdkerrors"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k8sutils "github.com/kong/gateway-operator/pkg/utils/kubernetes"
+
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+func createControlPlane(
+	ctx context.Context,
+	sdk *sdkkonnectgo.SDK,
+	logger logr.Logger, //nolint:unparam
+	cp *konnectv1alpha1.KonnectControlPlane,
+) error {
+	resp, err := sdk.ControlPlanes.CreateControlPlane(ctx, cp.Spec.CreateControlPlaneRequest)
+	// TODO: handle already exists
+	// Can't adopt it as it will cause conflicts between the controller
+	// that created that entity and already manages it, hm
+	// TODO: implement entity adoption https://github.com/Kong/gateway-operator/issues/460
```

Comment:
For the plugin API the SDK provides an `upsert` function. It'd be ideal to have the same function for other APIs as well

Reply:
👍 

Most if not all entity types have the support upsert in API. For those that do not have it we can submit requests.